src/main/kotlin/no/nav/tiltaksarrangor/koordinator/service/KoordinatorService.kt

```diff
@@ -63,12 +69,79 @@ class KoordinatorService(
 		)
 	}
 
-	fun getTilgjengeligeVeiledere(deltakerlisteId: UUID): List<TilgjengeligVeileder> {
-		return amtTiltakClient.getTilgjengeligeVeiledere(deltakerlisteId).map { it.toTilgjengeligVeileder() }
+	fun getTilgjengeligeVeiledere(deltakerlisteId: UUID, personIdent: String): List<TilgjengeligVeileder> {
+		val ansatt = getAnsattMedKoordinatorRoller(personIdent)
+		val deltakerliste = deltakerlisteRepository.getDeltakerliste(deltakerlisteId)
+			?: throw NoSuchElementException("Fant ikke deltakerliste med id $deltakerlisteId")
+
+		val harKoordinatorRolleHosArrangor = ansattService.harRolleHosArrangor(
+			arrangorId = deltakerliste.arrangorId,
+			rolle = AnsattRolle.KOORDINATOR,
+			roller = ansatt.roller
+		)
+
+		if (harKoordinatorRolleHosArrangor && ansattService.deltakerlisteErLagtTil(ansatt, deltakerlisteId)) {
+			return ansattService.getVeiledereForArrangor(deltakerliste.arrangorId).map {
+				TilgjengeligVeileder(
+					ansattId = it.id,
+					fornavn = it.fornavn,
+					mellomnavn = it.mellomnavn,
+					etternavn = it.etternavn
+				)
+			}
+		} else {
+			throw UnauthorizedException("Ansatt ${ansatt.id} har ikke tilgang til deltakerliste med id $deltakerlisteId")
+		}
+	}
+
+	fun tildelVeiledereForDeltaker(deltakerId: UUID, request: LeggTilVeiledereRequest, personIdent: String) {
+		val ansatt = getAnsattMedKoordinatorRoller(personIdent)
+		val deltakerMedDeltakerlisteDbo = deltakerRepository.getDeltakerMedDeltakerliste(deltakerId)
+			?: throw NoSuchElementException("Fant ikke deltaker med id $deltakerId")
+
+		val harKoordinatorRolleHosArrangor = ansattService.harRolleHosArrangor(
+			arrangorId = deltakerMedDeltakerlisteDbo.deltakerliste.arrangorId,
+			rolle = AnsattRolle.KOORDINATOR,
+			roller = ansatt.roller
+		)
+		if (harKoordinatorRolleHosArrangor && ansattService.deltakerlisteErLagtTil(ansatt, deltakerMedDeltakerlisteDbo.deltakerliste.id)) {
+			validerRequest(request)
+			if (!ansattService.erAlleAnsatteVeiledereHosArrangor(
+					ansattIder = request.veiledere.map { it.ansattId },
+					arrangorId = deltakerMedDeltakerlisteDbo.deltakerliste.arrangorId
+				)
+			) {
+				throw UnauthorizedException("Alle ansatte må ha veileder-rolle hos arrangør")
+			}
+			amtTiltakClient.tildelVeiledereForDeltaker(deltakerId, request)
```

Comment:
Flyten med at denne sender til både amt-arrangor og amt-tiltak virker rart hvis amt-arrangor nå også tar over veilederne. Hva mangler for at man kan skru av amt-tiltak her?

Reply:
Egentlig er det dette her som mangler for at vi kan la være å gjøre kallet til amt-tiltak :) Når bff-en gir beskjed til amt-arrangør om nye veiledere (og koordinators deltakerlister, som allerede er fikset) så har amt-arrangør nok informasjon til å kunne være master for hele ansatt-objektet. Da kan vi sette opp at amt-tiltak bare leser fra ansatt-topicen, og at amt-arrangør produserer til denne (i dag er det amt-tiltak som produserer til den topicen). Vi har en lapp på å gjøre den svitsjen.

Reply:
Ok, så bra! 🎉